### PR TITLE
Move screenshot button to calendar toolbar

### LINKED ISSUE #979: Screenshot button doesn't appear to work on mobile
## Description
I can't find the image that is supposed to be downloaded, and neither did I get a notification that a file was saved. This was the case on both Android PWA, Android Chrome browser, and Safari.

The button still works on desktop.

### PR DESCRIPTION
**Pull request by Douglas-Hong**
## Summary
- Moved the screenshot button from Export to the calendar toolbar, as discussed in last week's meeting. The screenshot button did not work on mobile devices because on small screen sizes, the Export button moves into the Settings menu. Since the Settings menu is on top of the main page, we cannot properly take a screenshot of the calendar.
- To save more space in the toolbar, the button to add custom events is now just a regular icon button. The "Custom" label was removed.

## Test Plan
- Click the screenshot button on desktop and verify that the schedule was downloaded successfully.
- Click the screenshot button on mobile viewports and verify that the schedule was downloaded succesfully.

## Issues

Closes #979

### CODE REVIEW

File: apps/antalmanac/src/components/Header/Export.tsx
Comment (KevinWu098) [inline apps/antalmanac/src/components/Header/Export.tsx:1]: Screenshot 2024-05-12 at 12 35 37 AM

issue: `Export` is now a dropdown with one option

suggestion: 
If there's only one functionality left in the export button, it would make more sense to just remove the `Popover` and make it a `Button`.

But, even better, I think we should move this into the Calendar Toolbar as an `IconButton` instead, since it applies specifically to the calendar.

File: apps/antalmanac/src/components/Calendar/CalendarToolbar.tsx
Comment (KevinWu098) [inline apps/antalmanac/src/components/Calendar/CalendarToolbar.tsx]:
```suggestion
            <Box display="flex" flexWrap="wrap" alignItems="center" gap={0.5}>
                <ScreenshotButton />
                    
                <Tooltip title="Undo last action">
                    <IconButton onClick={handleUndo} size="medium" disabled={skeletonMode}>
```

issue: it's not entirely in the diff, but we've got a bit of unnecessary code duplication for styling. 

suggestion:  remove the Boxes wrapping the buttons around all the buttons and have them in a singular Box, like shown above? It's all styling `gap={0.5}` and `flexWrap="wrap` anyways

This, basically:

```tsx
            <Box display="flex" flexWrap="wrap" alignItems="center" gap={0.5}>
                <ScreenshotButton />

                <Tooltip title="Undo last action">
                    <IconButton onClick={handleUndo} size="medium" disabled={skeletonMode}>
                        <UndoIcon fontSize="small" />
                    </IconButton>
                </Tooltip>

                <Tooltip title="Clear schedule">
                    <IconButton onClick={handleClearSchedule} size="medium" disabled={skeletonMode}>
                        <DeleteIcon fontSize="small" />
                    </IconButton>
                </Tooltip>

                <CustomEventDialog key="custom" scheduleNames={AppStore.getScheduleNames()} />
            </Box>
```

File: apps/antalmanac/src/components/Calendar/Toolbar/CustomEventDialog/CustomEventDialog.tsx
Comment (KevinWu098) [inline apps/antalmanac/src/components/Calendar/Toolbar/CustomEventDialog/CustomEventDialog.tsx]:
```suggestion
                    <IconButton onClick={handleOpen} size="medium" disabled={skeletonMode}>
```

issue: I know you didn't write this part, but `disableRipple` messes with the styling on hover and is inconsistent with the other buttons

File: apps/antalmanac/src/components/Calendar/Toolbar/CustomEventDialog/CustomEventDialog.tsx
Comment (KevinWu098) [inline apps/antalmanac/src/components/Calendar/Toolbar/CustomEventDialog/CustomEventDialog.tsx:1]: Screenshot 2024-05-12 at 12 42 13 AM

issue: `CustomEventDialog` is built with MUI v4, which causes styling inconsistencies w/ `Screenshot` and `CalendarToolbar`, both of which use MUI v5

suggestion: refactor like so:

```tsx
import { Add, Edit } from '@mui/icons-material';
import {
    Button,
    Dialog,
    DialogActions,
    DialogContent,
    FormControl,
    IconButton,
    Input,
    InputLabel,
    TextField,
    Tooltip,
} from '@mui/material';
```

**Review by KevinWu098 — CHANGES_REQUESTED** [review]
Thanks for writing up this PR! Very happy that we're continuing to cleanup AA's UI/UX :)

Just a few styling issues and an issue on `Export` (which is now a dropdown with a singular option) 🫡

**Comment by Douglas-Hong** [comment]
Thank you for the review, I might have been a bit careless with the copy pasting 💀 

After moving Export to the calendar toolbar, the header now only has Save, Load, Import, and the Settings bar. Since the header has more space, I tried keeping Import in the header on mobile screens instead of moving it to the Settings menu. We can discuss this change in tomorrow's meeting.

File: apps/antalmanac/src/components/buttons/ExportCalendar.tsx
Comment (KevinWu098) [inline apps/antalmanac/src/components/buttons/ExportCalendar.tsx:1]: nitpick (non-blocking): Might be more accurate if we renamed this file to `DownloadCalendar` (and any other associated refactoring). Or, keeping convention, just simply `Download.tsx` with an export of `DownloadButton`, like how `ScreenshotButton` is in `Screenshot.tsx`.

**Review by KevinWu098 — APPROVED** [review]
LGTM! Thanks for implementing the review suggestions -- I agree with moving Import back into the header on mobile, but let's get a final greenlight tomorrow before merging :)

**Review by KevinWu098 — APPROVED** [review]
LGTM 🚀🚀

**Pull request merged**
